createSystemUIFont: Remove unnecessary work around for empty locale

vitorroriz opened the pull request:
#### d89680f2dafd2a015683b990eee0d410d2dfa700
<pre>
createSystemUIFont: Remove unnecessary work around for empty locale
<a href="https://bugs.webkit.org/show_bug.cgi?id=304069">https://bugs.webkit.org/show_bug.cgi?id=304069</a>
<a href="https://rdar.apple.com/166389542">rdar://166389542</a>

Reviewed by NOBODY (OOPS!).

There used to be a difference between an empty and a null locale for CTFontCreateUIFontForLanguage.
Passing an empty string for locale would result in CoreText not using the system&apos;s locale.
As of <a href="https://rdar.apple.com/143372056">rdar://143372056</a> this is no longer true.

* Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
(WebCore::SystemFontDatabaseCoreText::createSystemUIFont):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d89680f2dafd2a015683b990eee0d410d2dfa700

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135727 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8103 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47023 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143442 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/87364 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ede38c73-0b46-47b8-9ca8-cdb0810be266) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137596 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8760 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/7950 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/103724 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/71759 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/597a01b8-3557-4bb0-9b5d-ef8b94fc6ea9) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138673 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6307 "Found 1 new test failure: fast/text/international/system-language/han-quotes.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121657 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84599 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0c87395a-fcfe-4fc7-851b-50299e244fcb) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/6065 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/3681 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4048 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/115291 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39844 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146191 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7794 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40412 "Found 1 new test failure: fast/text/international/system-language/han-quotes.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/112084 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7821 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/6538 "Found 1 new test failure: fast/text/international/system-language/han-quotes.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112464 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/5937 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/117957 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/61730 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/7839 "Built successfully") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/36064 "Found 1 new test failure: fast/text/international/system-language/han-quotes.html (failure)") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/7578 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71390 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/7802 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/7675 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
```diff
@@ -51,12 +51,6 @@ SystemFontDatabaseCoreText::SystemFontDatabaseCoreText() = default;
 
 RetainPtr<CTFontRef> SystemFontDatabaseCoreText::createSystemUIFont(const CascadeListParameters& parameters, CFStringRef locale)
 {
-    // Work around a quirk of the platform API.
```
I believe that the fix for this workaround only shipped in macOS 15.4/iOS 18.4. Since we build WebKit for the two prior releases (which still have this bug) we need to keep this.

Please use a compile guard here:

```
#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 150400)
...
#endif
```

brentfulgham reviewed (changes requested):
We don't want to regress down level WebKit/Safari, so please just wrap this workaround in a version check. We can remove this code once macOS older than 15.4 is no longer supported.

vitorroriz committed:
createSystemUIFont: Remove unnecessary work around for empty locale
https://bugs.webkit.org/show_bug.cgi?id=304069
rdar://166389542

Reviewed by NOBODY (OOPS!).

There used to be a difference between an empty and a null locale for CTFontCreateUIFontForLanguage.
Passing an empty string for locale would result in CoreText not using the system's locale.
As of rdar://143372056 this is no longer true.

* Source/WebCore/platform/graphics/cocoa/SystemFontDatabaseCoreText.cpp:
(WebCore::SystemFontDatabaseCoreText::createSystemUIFont):
